From e4c4d382cb10b876b9f37a68179203061713386c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pawe=C5=82=20Har=C5=82ozi=C5=84ski?=
 <pawel.harlozinski@intel.com>
Date: Tue, 4 Dec 2018 18:57:32 +0100
Subject: [PATCH 221/292] ASoC: Intel: Skylake: fixed not handling FW config
 info types
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Added parsing FW and HW info types: USB Audio Offload Link, Power Gating Policy

Change-Id: Ie92b78f4ae1bcba38d3f7b4d5297cfe59dc2d68b
Signed-off-by: Paweł Harłoziński <pawel.harlozinski@intel.com>
Reviewed-on:
Reviewed-by: Lewandowski, Gustaw <gustaw.lewandowski@intel.com>
---
 sound/soc/intel/skylake/skl-sst-dsp.h   | 3 +++
 sound/soc/intel/skylake/skl-sst-utils.c | 7 +++++++
 2 files changed, 10 insertions(+)

diff --git a/sound/soc/intel/skylake/skl-sst-dsp.h b/sound/soc/intel/skylake/skl-sst-dsp.h
index 01807675d7d9..e3287ee0daeb 100644
--- a/sound/soc/intel/skylake/skl-sst-dsp.h
+++ b/sound/soc/intel/skylake/skl-sst-dsp.h
@@ -162,6 +162,8 @@ enum skl_fw_info_type {
 	SKL_SCHEDULER_CONFIG,
 	SKL_XTAL_FREQ_HZ,
 	SKL_CLOCKS_CONFIG,
+	SKL_UAOL_SUPPORT,
+	SKL_POWER_GATING_POLICY,
 };
 
 enum skl_hw_info_type {
@@ -175,6 +177,7 @@ enum skl_hw_info_type {
 	SKL_HB_EBB_COUNT,
 	SKL_LP_EBB_COUNT,
 	SKL_EBB_SIZE_BYTES,
+	SKL_UAOL_CAPS_HW_CFG,
 };
 
 /* DSP Core state */
diff --git a/sound/soc/intel/skylake/skl-sst-utils.c b/sound/soc/intel/skylake/skl-sst-utils.c
index c4d573c92c1e..979cf44b5d06 100644
--- a/sound/soc/intel/skylake/skl-sst-utils.c
+++ b/sound/soc/intel/skylake/skl-sst-utils.c
@@ -658,6 +658,9 @@ static int skl_parse_hw_config_info(struct sst_dsp *ctx, u8 *src, int limit)
 			hw_property->ebb_size_bytes = *value;
 			break;
 
+		case SKL_UAOL_CAPS_HW_CFG:
+			break;
+
 		default:
 			dev_err(ctx->dev, "Invalid hw info type:%d \n", type);
 			break;
@@ -985,6 +988,10 @@ static int skl_parse_fw_config_info(struct sst_dsp *ctx,
 					message->length);
 			break;
 
+		case SKL_UAOL_SUPPORT:
+		case SKL_POWER_GATING_POLICY:
+			break;
+
 		default:
 			dev_err(ctx->dev, "Invalid fw info type:%d !!\n",
 					type);
-- 
2.21.0

